spec/java/util/LinkedHashSet.KeyIterator.lsl

```diff
@@ -88,12 +88,9 @@ automaton LinkedHashSet_KeyIteratorAutomaton
         if (!atValidPosition)
             action THROW_NEW("java.util.NoSuchElementException", []);
 
-        val key: Object = action SYMBOLIC("java.lang.Object");
-        action ASSUME(key != null);
-        action ASSUME(key != this.currentKey);
         val parentStorage: map<Object, Object> = LinkedHashSetAutomaton(this.parent).storage;
-        val sourceStorageHasKey: boolean = action MAP_HAS_KEY(parentStorage, key);
-        action ASSUME(sourceStorageHasKey);
+        val key: Object = action MAP_GET_ANY_KEY(parentStorage);
+        action ASSUME(key != this.currentKey);
         val dstStorageHasKey: boolean = action MAP_HAS_KEY(this.visitedKeys, key);
```

Comment:
Consider using supplementary `unseen: map<Object, Object>` to store items for later retrieval as suggested by Valentin.

Reply:
Fixed